pandas/core/sparse/series.py

```diff
@@ -628,14 +628,15 @@ def sparse_reindex(self, new_index):
                                  fill_value=self.fill_value).__finalize__(self)
 
     @Appender(generic._shared_docs['take'])
-    def take(self, indices, axis=0, convert=True, *args, **kwargs):
-        convert = nv.validate_take_with_convert(convert, args, kwargs)
-
-        if not convert:
+    def take(self, indices, axis=0, convert=-1, *args, **kwargs):
```

Comment:
same this is very non intuitive to use a placeholder value that is not valid

Reply:
Well, none is not a valid parameter for a boolean input either, and the compatibility issue still stands

Reply:
sure it is, you need to set the default *before* you check the inputs. That's the point.

```
if convert is not None:
     warnings.warn(.....)
      validate(convert)
```

Reply:
@jreback : I do set the default beforehand, and everything is green now.